server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -84,7 +85,9 @@
 
     private final AtomicLong taskIdGenerator = new AtomicLong();
 
-    private final Map<TaskId, String> banedParents = new ConcurrentHashMap<>();
+    private final Map<TaskId, BanReason> banedParents = new ConcurrentHashMap<>();
```

Comment:
I always wondered: should this be `bannedParents` ? If so, would now be a good time to rename it?

Reply:
Do you have any suggestion?

Reply:
yea I think it should be banned and not baned (double n). Not sure if renaming as part of your PR will add too much noise, up to you.

Reply:
Sure, will do. Thanks for your suggestion :).